examples/custom-service.mdx

```diff
@@ -20,49 +38,117 @@ edition = "2021"
 [lib]
 
 [dependencies]
-shuttle-service = { version = "0.9.0" }
-shuttle-shared-db = { version = "0.9.0", features = ["postgres"] }
-sqlx = { version = "0.6", features = ["runtime-tokio-native-tls", "postgres"] }
+anyhow = "1.0.62"
+axum = "0.6.4"
+hyper = "0.14.24"
+poise = "0.5.2"
+serde = "1.0"
+shuttle-secrets = "0.9.0"
+shuttle-service = { version = "0.9.0", features = ["bot-serenity"] }
+tokio = "1.22.0"
 ```
 
-This section will explain how to create a custom shuttle service. To do this we need 
-to return a struct from our `shuttle_service::main` function that implements `shuttle_service::Service`.
-
-In this simple example we will implement `Service` for a `PgPool`, that will simply return 'Hello world'
-from the database on startup. In Rust we can't implement foreign traits for foreign types, so we have 
-to create a wrapper around our `PgPool`.
+### Getting Started
+To get started, we need to return a wrapper struct from our `shuttle_service::main` function that 
+implements `shuttle_service::Service`, which we can declare like so:
 
 ```rust
-struct PoolService {
-    pool: sqlx::PgPool,
+struct CustomService {
+    discord_token: String,
 }
 ```
 
 Then we need to implement `shuttle_service::Service` for our wrapper. If you need to bind to an address,
 for example if you're implementing service for an HTTP server, you can use the `addr` argument from `bind`.
 You can only have one HTTP service bound to the `addr`, but you can start other services that don't rely on 
-binding to a socket. A common usecase is starting an HTTP server like `axum`, alongside a `serenity` bot 
-that doesn't need to bind to a socket. Then you would include both in the same wrapper struct, and start 
-them as normal in the `bind` function.
+binding to a socket.
+
+We can do do this by simply creating an Arc value from `self` and then cloning it every time we want to use it,
+and spawning another thread to run the service like so:
 
 ```rust lib.rs
 #[shuttle_service::async_trait]
-impl shuttle_service::Service for PoolService {
+impl shuttle_service::Service for CustomService {
     async fn bind(
         mut self: Box<Self>,
         _addr: std::net::SocketAddr,
     ) -> Result<(), shuttle_service::error::Error> {
-        self.start().await?;
+        let service = Arc::new(self);
+
+        let router = Arc::clone(&service);
+
+        tokio::spawn(async move {
+            Arc::clone(&service)
+                .set_up_bot()
+                .await
+                .expect("Something went wrong with the bot! :(");
+        });
+
+        router
+            .set_up_router(addr)
+            .await
+            .expect("Something went wrong with the router! :(");
```

Comment:
So if we wrap the poise client/framework, self will be that. So then we can do `self.start()`. I also think we should use `tokio::select!` here and start both services, the bot and axum, in that, rather than spawning the bot as a background task.

Reply:
Ah I didn't realise you could use `tokio::select!` for that - I'll include this in the next commit